src/test/scala/tech/cryptonomic/conseil/tezos/JsonDecodersTest.scala

```diff
@@ -39,6 +39,23 @@ class JsonDecodersTest extends WordSpec with Matchers with EitherValues {
 
     }
 
+    "decode null fields as empty Options" in {
```

Comment:
This test touches circe directly. It looks like we are missing some abstraction here.

Reply:
This is how I did it with Michelson parsing: https://github.com/Cryptonomic/Conseil/blob/master/src/main/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParser.scala#L208
Circe is used only internally and API doesn't expose it at all. I agree that it requires some code to be written but in my personal opinion, it's usually worth it. On the other hand, if refactor went smooth, maybe, in that case, it's not applicable. 

Anyway, my main point was that I'd prefer tests like "check if that production object with NULL field decodes correctly" rather than "let's check if that library with configuration copied from production code works correctly on some test object", or, even better: "check if this process works correctly if I pass NULL". The goal is to construct tests in a way to test behavior not implementation.

Reply:
I get your point, and this test is a regression check.

Previously it did the same verification using the Jackson library, and I ported the test to circe decoders once I switched to that, to preserve behaviour across implementations.

As you might see I actually tried to somehow keep things neatly separated 
 at least, by defining a `JsonDecoders.Circe` submodule, so that other JsonDecoders can be actually defined, using different library support.

In some sense the issue is that, while writing a Parser trait is pretty straight-forward, I didn't check how much additional work would be needed if we are to add encoding, too.

I'll think again about this, and we might possibly reuse your abstract approach everywhere.
I am missing time to think about this at the moment, so I'll just defer this task to some future moment